.github/workflows/pythonbuild.yml

```diff
@@ -135,91 +109,79 @@ jobs:
+          - tox-env: py39-apache
+          - tox-env: py310-apache
+          - tox-env: py311-apache
+          - tox-env: py312-apache
 
-          - python-version: "3.7"
-            tox-env: py37-azureblob
-          - python-version: "3.8"
-            tox-env: py38-azureblob
-          - python-version: "3.9"
-            tox-env: py39-azureblob
-          - python-version: "3.10"
-            tox-env: py310-azureblob
-          - python-version: "3.11"
-            tox-env: py311-azureblob
-          - python-version: "3.12"
-            tox-env: py312-azureblob
-
-
-          - python-version: "3.10"
-            tox-env: flake8
-          - python-version: "3.10"
-            tox-env: docs
+          - tox-env: py38-azureblob
+          - tox-env: py39-azureblob
+          - tox-env: py310-azureblob
+          - tox-env: py311-azureblob
+          - tox-env: py312-azureblob
 
     steps:
       - uses: actions/checkout@v4
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v5
+      - name: Set up the latest version of uv
+        uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+          cache-dependency-glob: "pyproject.toml"
+      - name: Install dependencies
+        run: |
+          uv tool install --python-preference only-managed --python 3.12 tox --with tox-uv
+      - name: Build
+        env:
+          TOXENV: ${{ matrix.tox-env }}
+          OVERRIDE_SKIP_CI_TESTS: ${{ matrix.OVERRIDE_SKIP_CI_TESTS }}
+        run: uvx --with tox-uv tox run
+      - name: Upload coverage to Codecov
+        if: ${{ matrix.tox-env != 'flake8' && matrix.tox-env != 'docs' }}
+        uses: codecov/codecov-action@v5
         with:
-          python-version: ${{ matrix.python-version }}
-      - name: Cache pip
-        uses: actions/cache@v4
+          fail_ci_if_error: true
+          verbose: true
+  others:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        include:
+          - tox-env: flake8
+          - tox-env: docs
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up the latest version of uv
+        uses: astral-sh/setup-uv@v5
         with:
-          # This path is specific to Ubuntu
-          path: ~/.cache/pip
-          # Look to see if there is a cache hit for the corresponding requirements files
-          key: ${{ format('{0}-pip-{1}', runner.os, hashFiles('dev-requirements.txt', format('requirements{0}.txt', matrix.spark-version-suffix))) }}
+          enable-cache: true
+          cache-dependency-glob: "pyproject.toml"
       - name: Install dependencies
         run: |
-          python -m pip install --upgrade pip 'tox<4.0'
+          uv tool install --python-preference only-managed --python 3.12 tox --with tox-uv
       - name: Build
         env:
           TOXENV: ${{ matrix.tox-env }}
           OVERRIDE_SKIP_CI_TESTS: ${{ matrix.OVERRIDE_SKIP_CI_TESTS }}
-        run: tox
+        run: uvx --with tox-uv tox run
```

Comment:
The step just below this line includes a conditional to not run coverage upload if flake8 or docs. But this "others" matrix only includes flake8 and docs. Could we then remove its `Upload coverage to Codecov` step entirely?

Reply:
fixed in [48650b7](https://github.com/spotify/luigi/pull/3338/commits/48650b77e5be8b53454a8170e19179645f309f09)